src/System.Text.Json/src/System/Text/Json/Serialization/ReflectionEmitMemberAccessor.cs

```diff
@@ -16,6 +16,11 @@ public override JsonClassInfo.ConstructorDelegate CreateConstructor(Type type)
             Debug.Assert(type != null);
             ConstructorInfo realMethod = type.GetConstructor(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance, binder: null, Type.EmptyTypes, modifiers: null);
 
+            if (type.IsAbstract)
```

Comment:
If you're not going to put this before `GetConstructor`, then this can be merged with the next check.

```csharp
Debug.Assert(type != null);

if (type.IsAbstract)
{
    return null;
}

ConstructorInfo realMethod = type.GetConstructor(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance, binder: null, Type.EmptyTypes, modifiers: null);

if (realMethod == null && !type.IsValueType)
{
    return null;
}
```
```csharp
if (realMethod == null && !type.IsValueType || type.IsAbstract)
```

Reply:
I split it out for debugging.